src/components/Modal/ModalContainer.test.tsx

```diff
@@ -49,3 +54,46 @@ it('should provide a ref to the div html element', () => {
   render(<ModalContainer ref={ref}>Container</ModalContainer>)
   expect(ref.current.tagName).toEqual('DIV')
 })
+
+describe('conditional rendering and prop passing to ModalHeader', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('does not render ModalHeader when title is not provided', () => {
+    render(<ModalContainer>Container content</ModalContainer>)
+
+    expect(ModalHeader).not.toHaveBeenCalled()
+  })
+
+  it('passes all expected props correctly to ModalHeader', () => {
+    const onClose = jest.fn()
+
+    render(
+      <ModalContainer
+        title='title'
+        subtitle='subtitle'
+        headerIcon='infoCircleOutline'
+        headerBackground='red'
+        showHeaderBorder={true}
+        hasCloseIcon={true}
```

Comment:
```suggestion
        showHeaderBorder
        hasCloseIcon
```

You don't need to specify the `true`